core/src/main/java/com/linecorp/armeria/client/HttpSessionHandler.java

```diff
@@ -161,7 +164,7 @@ public boolean incrementNumUnfinishedResponses() {
     @Override
     public boolean canSendRequest() {
         assert responseDecoder != null;
-        return active && !responseDecoder.needsToDisconnectWhenFinished();
+        return active && !responseDecoder.needsToDisconnectNow();
```

Comment:
I am not sure that this change is ok.
Also I checked other PR is improving 'canSendRequest' method.
https://github.com/line/armeria/pull/4531/files#diff-3c7269b1753aa13e46bd5490cb6087cd4175bcc6cfdf8686a7fe4f81ab5a4aaaR167

Reply:
I make new `canNextRequest` method to check accepting next request.